Source/missiles.cpp

```diff
@@ -3037,18 +3033,18 @@ void ProcessFireball(Missile &missile)
 			}
 
 			if (!TransList[dTransVal[missilePosition.x][missilePosition.y]]
-			    || (missile.position.velocity.deltaX < 0 && ((TransList[dTransVal[missilePosition.x][missilePosition.y + 1]] && TileHasAny(dPiece[missilePosition.x][missilePosition.y + 1], TileProperties::Solid)) || (TransList[dTransVal[missilePosition.x][missilePosition.y - 1]] && TileHasAny(dPiece[missilePosition.x][missilePosition.y - 1], TileProperties::Solid))))) {
-				missile.position.tile += Displacement { 1, 1 };
+			    || (missile.position.velocity.deltaX < 0 && ((TransList[dTransVal[missilePosition.x][missilePosition.y + 1]] && TileHasAny(missilePosition + Direction::SouthEast, TileProperties::Solid)) || (TransList[dTransVal[missilePosition.x][missilePosition.y - 1]] && TileHasAny(missilePosition + Direction::NorthWest, TileProperties::Solid))))) {

```

Comment:
Ok, I think I got it right this time, you can check them here: https://github.com/diasurgical/devilutionX/blob/fc6f9edad3daa3460106d8df5c67148cacffbeaf/Source/engine/displacement.hpp#L248
```suggestion
			    || (missile.position.velocity.deltaX < 0 && ((TransList[dTransVal[missilePosition.x][missilePosition.y + 1]] && TileHasAny(missilePosition + Direction::SouthWest, TileProperties::Solid)) || (TransList[dTransVal[missilePosition.x][missilePosition.y - 1]] && TileHasAny(missilePosition + Direction::NorthEast, TileProperties::Solid))))) {
```

Reply:
I'm not so sure this has to do with rendering, but maybe more collision... i might be wrong, maybe remove it and see what happens.